source/language/classical.rst

```diff
@@ -1,17 +1,18 @@
 Classical instructions
 ======================
 
-We envision two levels of classical control that we call low-level
-instructions and high-level kernel functions. Simple, fast instructions
-control the flow of the program and allow basic computations on
-lower-level parallel control processors. These instructions are likely
+We envision two levels of classical control: simple, low-level instructions embedded as
+part of a quantum circuit and high-level external functions which perform more complex
+classical computations. The low-level functions are fast, apply control flow and allow basic
+computations on lower-level parallel control processors. These instructions are likely
 to have known durations and many such instructions might be executed
-within the qubit coherence time. High-level kernel functions execute
-arbitrary user-defined classical subroutines that may be neither fast
-nor guaranteed to return. These assume a mechanism for passing data to
-and from higher-level processors. The kernel functions run on the global
-processor concurrently with operations on the local processors, if
-possible. Kernel functions can write to the global controller’s memory,
+within the qubit coherence time. The external, or ``extern``, functions execute
+complex blocks of classical code that may be neither fast nor guaranteed to return. In order
+to connect with classical compilation infrastucture, ``extern`` functions are defined outside of
+quantum circuits. The compiler toolchain is expected to link ``extern`` functions when building an
+executable. This strategy allows the programmer to use existing libraries with porting them into
```

Comment:
```suggestion
executable. This strategy allows the programmer to use existing libraries without porting them into
```

Reply:
Addressed in https://github.com/Qiskit/openqasm/pull/204/commits/a18f2cb9de95408601a0a1abf9c3ae1caccd5497.